storage/range.go

```diff
@@ -1962,5 +1955,7 @@ func (r *Range) ChangeReplicas(changeType proto.ReplicaChangeType, replica proto
 	if err != nil {
 		return util.Errorf("change replicas of %d failed: %s", desc.RaftID, err)
 	}
+	// Enqueue this range again to see if there are more changes to be made.
+	r.rm.ReplicateQueue().MaybeAdd(r, r.rm.Clock().Now())
```

Comment:
Just call MaybeAdd in a goroutine from the replication queue. Putting it here feels like the abstraction is bleeding. Why should this code know about the replication queue? This is something called from the replication queue (but maybe not in other cases). Also, the code in the replicate queue which causes this function to be run actually knows whether or not MaybeAdd ought to be called in the first place. We shouldn't just be calling MaybeAdd on queues willy-nilly. I think we should have a clear idea of why we're calling it.

I like to minimize calls to add to queues outside of the scanner.